### PR TITLE
Add support for FETCH FIRST xxx ROWS ONLY

### PR DESCRIPTION
**Pull request by matriv**
## Summary of the changes / Why this improves CrateDB

- Allow `OFFSET` and `LIMIT` clauses to be declared in any order, i.e.:
  ```
  SELECT * FROM t LIMIT 10 OFFSET 5
  ```
  or
  ```
  SELECT * FROM t OFFSET 5 LIMIT 10
  ```
  This improves compatibility with PostgreSQL.

- Add support for FETCH FIRST xxx ROWS ONLY.
  To improve compatibility with PostgreSQL, add support for the
  `FETCH [FIRST | NEXT] <noRows> [ROW|ROWS] ONLY` statement, which
  is an alternative of `LIMIT <noRows>`.

  Closes: https://github.com/crate/crate/issues/11540

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: libs/sql-parser/src/main/antlr/SqlBase.g4 [hunk 1]
```diff
@@ -134,8 +134,15 @@ query
 queryNoWith
     : queryTerm
       (ORDER BY sortItem (',' sortItem)*)?
-      (LIMIT (limit=parameterOrInteger | ALL))?
-      (OFFSET offset=parameterOrInteger (ROW | ROWS)?)?
+      (limitClause? offsetClause? | offsetClause? limitClause?)?
```
Comment (jeeminso) [inline libs/sql-parser/src/main/antlr/SqlBase.g4:137]: I think this could result in consecutive offsets or limits, `select * from t limit 10 limit 10`

Comment (matriv) [reply libs/sql-parser/src/main/antlr/SqlBase.g4:137]: I can add this test, but it's not:
```
@Test
public void testParseErrorMultipleLimits() {
    assertThrowsMatches(
        () -> SqlParser.createStatement("select fuu from dual limit 1 limit 2 limit 3"),
        ParsingException.class,
        "line 1:30: mismatched input 'limit' expecting {<EOF>, ';'}");
    assertThrowsMatches(
        () -> SqlParser.createStatement("select fuu from dual fetch first 1 rows only fetch first 2 rows only"),
        ParsingException.class,
        "line 1:46: mismatched input 'fetch' expecting {<EOF>, ';'}");
}
```

File: docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -74,6 +74,13 @@ None
 Changes
 =======
 
+- Added support for ``FETCH [FIRST | NEXT] <noRows> [ROW | ROWS] ONLY clause as
```
Comment (jeeminso) [inline docs/appendices/release-notes/unreleased.rst:77]:
```suggestion
- Added support for ``FETCH [FIRST | NEXT] <noRows> [ROW | ROWS] ONLY`` clause as
```

File: docs/appendices/release-notes/unreleased.rst [hunk 3]
```diff
@@ -74,6 +74,13 @@ None
...
+  and alternative to the ``LIMIT`` clause.
+
+- Allow ``LIMIT`` and ``OFFSET`` clauses to be declared in any order, i.e.:
```
Comment (jeeminso) [inline docs/appendices/release-notes/unreleased.rst:80]:
```suggestion
- Allowed ``LIMIT`` and ``OFFSET`` clauses to be declared in any order, i.e.:
```

Comment (jeeminso) [reply docs/appendices/release-notes/unreleased.rst:80]: just to be consistent

**Review by jeeminso — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
retest this please

File: libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 4]
```diff
@@ -1191,7 +1196,9 @@ public void testInsertStmtBuilder() {
         // insert from query
         printStatement("insert into foo (id, name) select id, name from bar order by id");
         printStatement("insert into foo (id, name) select * from bar limit 3 offset 10");
+        printStatement("insert into foo (id, name) select * from bar fetch first 3 rows only offset 10");
```
[the remark this answers is not in the document]
Comment (matriv) [reply libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:1199]: I'll just leave one here, just for completeness

**Pull request merged**